sirepo/auth/__init__.py

```diff
@@ -402,13 +402,26 @@ def reset_state():
 
 
 @contextlib.contextmanager
-def set_user_outside_of_http_request(uid, method='guest'):
-    """A user set explicitly outside of flask request cycle"""
+def set_user_outside_of_http_request(uid):
+    """A user set explicitly outside of flask request cycle
+
+    This will guess the auth method the user used to authenticate.
+    If the method cannot be guessed it will default to guest if configured
+    otherwise raise.
+    """
+    def _auth_module():
+        for m in _METHOD_MODULES.values():
```

Comment:
this list should be cfg.methods, that is, the methods that are configured but not deprecated and possibly hidden (basic is not visible).

Reply:
Wouldn't we want `visible_methods` not `cfg.methods`? `cfg.methods` may contain not visible methods.

Reply:
nvm I see what your are saying. We should allow non-visible methods for this. Implementing...

Reply:
1f427b40